src/live/src/user-to-user/live-service.js

```diff
@@ -0,0 +1,335 @@
+
+/**
+ * @typedef LiveServiceReceiver
+ * @property {Function} onMessage
+ * @property {Function} onError
+ * @property {Function} onStatus
+ */
+
+class LiveService {
+  /** @type {Client} @private */
+  _client;
+  /** @type {PubNub} @private */
+  _pubnubClient;
+  /** @type {PubNubListener} @private */
+  _pubnubListener;
+  /** @type {string} @private */
+  _userChannelGroup;
+  /** @type {Object} @private */
+  _pubnubConfig
+  /** @type {User} @private */
+  _registeredUser;
+
+  /**
+   * @constructor
+   * @param {Client} client
+   */
+  constructor(client) {
+    this._client = client || Client.sharedInstance();
+  }
+
+  get _pubnubClient() {
+    if (!this.__pubnubClient) {
+      this._throwNotInitializedError();
+    }
+    return this.__pubnubClient;
+  }
+
+  set _pubnubClient(value) {
+    this.__pubnubClient = value;
+  }
+
+  set _pubnubListener(value) {
+    this.__pubnubListener = value;
+  }
+
+  get _pubnubListener() {
+    if (!this.__pubnubListener) {
+      this._throwNotInitializedError();
+    }
+    return this.__pubnubListener;
+  }
+
+  isInitialized() {
+    return !!this.__pubnubClient && !!this.__pubnubListener && !!this._registeredUser;
+  }
+
+  /**
+   * Registers the active user for live service
+   * @returns {Promise}
+   */
+  registerUser(user) {
+    if (!user || !user.isActive()) {
+      return Promise.reject(new ActiveUserError('Missing or invalid active user'));
+    }
+
+    return this._makeRegisterRequest(user._id)
+      .then((regResponse) => {
+        this._registeredUser = user;
+        this._userChannelGroup = regResponse.userChannelGroup;
+        const config = extend({
+          ssl: true,
+          authKey: this._registeredUser._kmd.authtoken
+        }, regResponse);
+        return config;
+      });
+  }
+
+  /**
+   * @param {PubNub} pubnubClient
+   * @param {PubNubListener} pubnubListener
+   */
+  initialize(pubnubClient, pubnubListener) {
+    this._pubnubListener = pubnubListener;
+    this._pubnubClient = pubnubClient;
+    this._pubnubClient.addListener(this._pubnubListener);
+    this._subscribeToUserChannelGroup();
+  }
+
+  /**
+   * Unsubscribes from all events in PubNub client and in listener.
+   * Unregisters user from live service
+   */
+  uninitialize() {
+    this.unsubscribeFromAll();
+    this._pubnubClient = null;
+    this._pubnubListener = null;
+  }
+
+  onConnectionStatusUpdates(func) {
+    this._pubnubListener.on(PubNubListener.unclassifiedEvents, func);
+  }
+
+  offConnectionStatusUpdates(func) {
+    if (func) {
+      this._pubnubListener.removeListener(PubNubListener.unclassifiedEvents, func);
+    } else {
+      this._pubnubListener.removeAllListeners(PubNubListener.unclassifiedEvents);
+    }
+  }
+
+  /**
+   * @param {string} channelName The name of the PubNub channel to publish to
+   * @param {Object} message The message to be published
+   * @returns {Promise}
+   */
+  publishToChannel(channelName, message) {
```

Comment:
Depends on what you are referring to as "public". In order to be used by `Stream`, `LiveService` should expose this as public, in terms of data hiding. But the `LiveServiceFacade` hides this public method of the class from the end user, so they cannot call it. That's the point of the facade. Perhaps it needs a better name, since now it features a bit of logic (the `registerUser()` to `initialize()`). I'm a little unsure about this part.